src/lib/Key.js

```diff
@@ -91,6 +161,124 @@ class Key {
             : this._secret;
     }
 
+    /**
+     * @param {EncryptionKeyParams} keyParams
+     * @returns {Promise<CryptoKey>}
+     */
+    async getRsaPrivateKey(keyParams) {
+        if (!this.rsaKeyPair || !Key._isEncryptionKeyParamsEqual(keyParams, this.rsaKeyPair.keyParams)) {
+            this.rsaKeyPair = await this._computeRsaKeyPair(keyParams);
+            if (Key._isDefaultEncryptionKeyParams(keyParams)) {
+                await KeyStore.instance.setRsaKeypair(this.id, this.rsaKeyPair);
+            }
+        }
+
+        return window.crypto.subtle.importKey(
+            'pkcs8',
+            this.rsaKeyPair.privateKey,
+            { name: 'RSA-OAEP', hash: 'SHA-256' },
+            false, // Prevent extraction
+            ['decrypt'],
+        );
+    }
+
+    /**
+     * @param {EncryptionKeyParams} keyParams
+     * @returns {Promise<CryptoKey>}
+     */
+    async getRsaPublicKey(keyParams) {
+        if (!this.rsaKeyPair || !Key._isEncryptionKeyParamsEqual(keyParams, this.rsaKeyPair.keyParams)) {
+            this.rsaKeyPair = await this._computeRsaKeyPair(keyParams);
+            if (Key._isDefaultEncryptionKeyParams(keyParams)) {
+                await KeyStore.instance.setRsaKeypair(this.id, this.rsaKeyPair);
+            }
+        }
```

Comment:
This little snippet could be deduplicated into a method `setRsaKeyPair`.
```suggestion
        const rsaKeyPair = await this.setRsaKeyPair(keyParams);
```

With method:
```js
    /**
     * Calculates and sets the RSA key pair for the given key params. The RSA key is calculated deterministically based
     * on _secret, such that the same RSA key is generated for the same key params.
     * Warning: this method is computationally expensive.
     * @param {EncryptionKeyParams} keyParams
     * @returns {Promise<RsaKeyPairExport>}
     */
    async setRsaKeyPair(keyParams) {
        if (!this._rsaKeyPair || !Key._areEncryptionKeyParamsEqual(keyParams, this._rsaKeyPair.keyParams)) {
            this._rsaKeyPair = await this._computeRsaKeyPair(keyParams);
            if (Key._areDefaultEncryptionKeyParams(keyParams)) {
                await KeyStore.instance.setRsaKeypair(this.id, this._rsaKeyPair);
            }
        }
        return this._rsaKeyPair;
    }
```

Reply:
Resolved in #524